libs/@blockprotocol/graph/src/shared/stdlib/subgraph/edge/shared.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  BaseUrl,
+  extractBaseUrl,
+  extractVersion,
+  VersionedUrl,
+} from "@blockprotocol/type-system/slim";
+
+import {
+  OntologyOutwardEdge,
+  OntologyTypeRevisionId,
+  OntologyTypeVertexId,
+  OutwardEdge,
+  StrictOntologyOutwardEdge,
+  Subgraph,
+} from "../../../types/subgraph.js";
+
+/**
+ * This is a helper function to extract ontology edges given an `outwardEdgePredicate` function.
+ *
+ * @internal
+ * @param subgraph {Subgraph} - The `Subgraph` containing the underlying ontology types
+ * @param ontologyTypeId {OntologyTypeVertexId | VersionedUrl} - The identifier of the `EntityType` to search for
+ * @param outwardEdgePredicate {(outwardEdge: OutwardEdge<boolean>) => outwardEdge is StrictOntologyOutwardEdge} - The predicate to filter edges by
+ * @returns {OntologyTypeVertexId[]} - The resulting endpoints of the filtered edges
+ */
+export const getOntologyTypesReferencedBy = (
+  subgraph: Subgraph<boolean>,
+  ontologyTypeId: OntologyTypeVertexId | VersionedUrl,
+  outwardEdgePredicate: (
+    outwardEdge: OutwardEdge<boolean>,
+  ) => outwardEdge is StrictOntologyOutwardEdge,
```

Comment:
![image](https://user-images.githubusercontent.com/6988668/224661976-f93ba6c9-5b60-49be-a0a5-a44eab887351.png)

It gives a type error because the endpoint is incompatible
(`StrictOntologyOutwardEdge` has been renamed to `OntologyToOntologyOutwardEdge`)

Reply:
Nice :)